src/devices/machine/i8291a.h

```diff
@@ -0,0 +1,415 @@
+// license:BSD-3-Clause
+// copyright-holders: Sven Schnelle
+/*********************************************************************
+
+    i8291a.h
+
+    I8291A GPIB controller
+
+*********************************************************************/
+
+#ifndef MAME_MACHINE_I8291A_H
+#define MAME_MACHINE_I8291A_H
+
+
+#pragma once
+
+
+#define MCFG_I8291A_INT_WRITE_CB(_write)                               \
+	downcast<i8291a_device &>(*device).set_int_write_cb(DEVCB_##_write);
+
+#define MCFG_I8291A_DREQ_WRITE_CB(_write)                               \
+	downcast<i8291a_device &>(*device).set_dreq_write_cb(DEVCB_##_write);
+
+#define MCFG_I8291A_TRIG_WRITE_CB(_write)                               \
+	downcast<i8291a_device &>(*device).set_trig_write_cb(DEVCB_##_write);
+
+#define MCFG_I8291A_DIO_READ_CB(_read)             \
+	downcast<i8291a_device &>(*device).set_dio_read_cb(DEVCB_##_read);
+
+#define MCFG_I8291A_DIO_WRITE_CB(_write)                   \
+	downcast<i8291a_device &>(*device).set_dio_write_cb(DEVCB_##_write);
+
+#define MCFG_I8291A_EOI_WRITE_CB(_write)                               \
+	downcast<i8291a_device &>(*device).set_eoi_write_cb(DEVCB_##_write);
+
+#define MCFG_I8291A_DAV_WRITE_CB(_write)                               \
+	downcast<i8291a_device &>(*device).set_dav_write_cb(DEVCB_##_write);
+
+#define MCFG_I8291A_NRFD_WRITE_CB(_write)                              \
+	downcast<i8291a_device &>(*device).set_nrfd_write_cb(DEVCB_##_write);
+
+#define MCFG_I8291A_NDAC_WRITE_CB(_write)                              \
+	downcast<i8291a_device &>(*device).set_ndac_write_cb(DEVCB_##_write);
+
+#define MCFG_I8291A_SRQ_WRITE_CB(_write)                               \
+	downcast<i8291a_device &>(*device).set_srq_write_cb(DEVCB_##_write);
+
+
+class i8291a_device : public device_t
+{
+public:
+	// construction/destruction
+	i8291a_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock);
+
+	template <class Object> devcb_base& set_int_write_cb(Object &&cb)
+		{ return m_int_write_func.set_callback(std::forward<Object>(cb)); }
```

Comment:
Please follow convention in the rest of the source and don't indent the opening brace itself.  See e.g. `src/emu/addrmap.h` for an example of this style of inline function in a header.